src/Turtle/Options.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-- TODO: documentation
+
+module Turtle.Options
+    ( Parser
+    , ParameterName(..)
+    , LongName(..)
+    , HelpMessage(..)
+    , options
+    , switch
+    , parameter
+    , parameterAuto
+    , parameterText
+    , parameterInteger
+    , parameterInt
+    , parameterDouble
```

Comment:
What about `ParameterName`? Should it be renamed to `ArgumentName`?

Also I'm not sure about `argAuto`. Perhaps `argRead` would be more descriptive because we're using the `Read` instance.